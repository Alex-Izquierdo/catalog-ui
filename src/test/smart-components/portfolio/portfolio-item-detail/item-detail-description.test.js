import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MemoryRouter, Route } from 'react-router-dom';

import ItemDetailDescription from '../../../../SmartComponents/Portfolio/portfolio-item-detail/item-detail-description';

describe('<ItemDetailDescription />', () => {
  let initialProps;

  beforeEach(() => {
    initialProps = {
      product: {
        description: 'Product description',
        long_description: 'Product long description',
        support_url: 'some/support/url',
        documentation_url: 'some/documention/url'
      },
      url: '/base/url',
      workflows: [{
        label: 'Foo',
        value: 'foo'
      }, {
        label: 'Bar',
        value: null
      }],
      workflow: 'foo',
      setWorkflow: jest.fn()
    };
  });

  it('should render correctly', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={ [ '/base/url' ] }>
        <Route path="/base/url" render={ () => <ItemDetailDescription { ...initialProps } /> } />
      </MemoryRouter>
    );
    expect(toJson(wrapper.find(ItemDetailDescription))).toMatchSnapshot();
  });

  it('should render correctly in edit variant', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={ [ '/base/url/edit' ] }>
        <Route path="/base/url/edit" render={ () => <ItemDetailDescription { ...initialProps } /> } />
      </MemoryRouter>
    );
    expect(toJson(wrapper.find(ItemDetailDescription))).toMatchSnapshot();
  });
});
