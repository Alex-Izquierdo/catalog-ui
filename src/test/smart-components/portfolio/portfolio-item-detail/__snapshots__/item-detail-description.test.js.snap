// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemDetailDescription /> should render correctly 1`] = `
<ItemDetailDescription
  product={
    Object {
      "description": "Product description",
      "documentation_url": "some/documention/url",
      "long_description": "Product long description",
      "support_url": "some/support/url",
    }
  }
  setWorkflow={[MockFunction]}
  url="/base/url"
  workflow="foo"
  workflows={
    Array [
      Object {
        "label": "Foo",
        "value": "foo",
      },
      Object {
        "label": "Bar",
        "value": "bar",
      },
    ]
  }
>
  <TextContent
    className=""
  >
    <div
      className="pf-c-content"
    >
      <Text
        className=""
        component="h6"
      >
        <h6
          className=""
          data-pf-content={true}
        >
          Overview
        </h6>
      </Text>
      <Text
        className=""
        component="p"
      >
        <p
          className=""
          data-pf-content={true}
        >
          Product description
        </p>
      </Text>
      <Text
        className=""
        component="p"
      >
        <p
          className=""
          data-pf-content={true}
        >
          Product long description
        </p>
      </Text>
      <Text
        className=""
        component="p"
      >
        <p
          className=""
          data-pf-content={true}
        >
          <a
            href="some/support/url"
            rel="noopener noreferrer"
            target="_blank"
          >
            Learn more
          </a>
        </p>
      </Text>
      <Text
        className=""
        component="h6"
      >
        <h6
          className=""
          data-pf-content={true}
        >
          Documentation
        </h6>
      </Text>
      <Text
        className=""
        component="p"
      >
        <p
          className=""
          data-pf-content={true}
        >
          <a
            href="some/documention/url"
            rel="noopener noreferrer"
            target="_blank"
          >
            Doc link
          </a>
        </p>
      </Text>
      <Route
        exact={true}
        path="/base/url"
        render={[Function]}
      >
        <Text
          className=""
          component="h6"
        >
          <h6
            className=""
            data-pf-content={true}
          >
            Approval workflow
          </h6>
        </Text>
        <Text
          className=""
          component="p"
        >
          <p
            className=""
            data-pf-content={true}
          >
            None
          </p>
        </Text>
      </Route>
    </div>
  </TextContent>
  <Route
    exact={true}
    path="/base/url/edit"
    render={[Function]}
  />
</ItemDetailDescription>
`;

exports[`<ItemDetailDescription /> should render correctly in edit variant 1`] = `
<ItemDetailDescription
  product={
    Object {
      "description": "Product description",
      "documentation_url": "some/documention/url",
      "long_description": "Product long description",
      "support_url": "some/support/url",
    }
  }
  setWorkflow={[MockFunction]}
  url="/base/url"
  workflow="foo"
  workflows={
    Array [
      Object {
        "label": "Foo",
        "value": "foo",
      },
      Object {
        "label": "Bar",
        "value": "bar",
      },
    ]
  }
>
  <TextContent
    className=""
  >
    <div
      className="pf-c-content"
    >
      <Text
        className=""
        component="h6"
      >
        <h6
          className=""
          data-pf-content={true}
        >
          Overview
        </h6>
      </Text>
      <Text
        className=""
        component="p"
      >
        <p
          className=""
          data-pf-content={true}
        >
          Product description
        </p>
      </Text>
      <Text
        className=""
        component="p"
      >
        <p
          className=""
          data-pf-content={true}
        >
          Product long description
        </p>
      </Text>
      <Text
        className=""
        component="p"
      >
        <p
          className=""
          data-pf-content={true}
        >
          <a
            href="some/support/url"
            rel="noopener noreferrer"
            target="_blank"
          >
            Learn more
          </a>
        </p>
      </Text>
      <Text
        className=""
        component="h6"
      >
        <h6
          className=""
          data-pf-content={true}
        >
          Documentation
        </h6>
      </Text>
      <Text
        className=""
        component="p"
      >
        <p
          className=""
          data-pf-content={true}
        >
          <a
            href="some/documention/url"
            rel="noopener noreferrer"
            target="_blank"
          >
            Doc link
          </a>
        </p>
      </Text>
      <Route
        exact={true}
        path="/base/url"
        render={[Function]}
      />
    </div>
  </TextContent>
  <Route
    exact={true}
    path="/base/url/edit"
    render={[Function]}
  >
    <Grid
      className=""
      gutter={null}
      lg={null}
      md={null}
      sm={null}
      span={null}
      xl={null}
    >
      <div
        className="pf-l-grid"
      >
        <GridItem
          className=""
          lg={null}
          lgOffset={null}
          lgRowSpan={null}
          md={6}
          mdOffset={null}
          mdRowSpan={null}
          offset={null}
          rowSpan={null}
          sm={null}
          smOffset={null}
          smRowSpan={null}
          span={null}
          xl={null}
          xlOffset={null}
          xlRowSpan={null}
        >
          <div
            className="pf-l-grid__item pf-m-6-col-on-md"
          >
            <Pf4SelectWrapper
              id="change-workflow"
              input={
                Object {
                  "onChange": [Function],
                  "value": "foo",
                }
              }
              label="Approval workflow"
              meta={Object {}}
              options={
                Array [
                  Object {
                    "label": "Foo",
                    "value": "foo",
                  },
                  Object {
                    "label": "Bar",
                    "value": "bar",
                  },
                ]
              }
            >
              <FormGroup
                className=""
                fieldId="change-workflow"
                isInline={false}
                isRequired={false}
                isValid={true}
                label="Approval workflow"
              >
                <div
                  className="pf-c-form__group"
                >
                  <label
                    className="pf-c-form__label"
                    htmlFor="change-workflow"
                  >
                    Approval workflow
                  </label>
                  <Select
                    formOptions={Object {}}
                    id="change-workflow"
                    input={
                      Object {
                        "onChange": [Function],
                        "value": "foo",
                      }
                    }
                    isValid={true}
                    label="Approval workflow"
                    options={
                      Array [
                        Object {
                          "label": "Foo",
                          "value": "foo",
                        },
                        Object {
                          "label": "Bar",
                          "value": "bar",
                        },
                      ]
                    }
                  >
                    <FormSelect
                      aria-label={null}
                      className=""
                      id="change-workflow"
                      isDisabled={false}
                      isValid={true}
                      label="Approval workflow"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      value="foo"
                    >
                      <select
                        aria-invalid={false}
                        aria-label={null}
                        className="pf-c-form-control"
                        disabled={false}
                        id="change-workflow"
                        label="Approval workflow"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        value="foo"
                      >
                        <FormSelectOption
                          className=""
                          isDisabled={false}
                          key="None"
                          label="None"
                          value=""
                        >
                          <option
                            className=""
                            disabled={false}
                            value=""
                          >
                            None
                          </option>
                        </FormSelectOption>
                        <FormSelectOption
                          className=""
                          isDisabled={false}
                          key="foo"
                          label="Foo"
                          value="foo"
                        >
                          <option
                            className=""
                            disabled={false}
                            value="foo"
                          >
                            Foo
                          </option>
                        </FormSelectOption>
                        <FormSelectOption
                          className=""
                          isDisabled={false}
                          key="bar"
                          label="Bar"
                          value="bar"
                        >
                          <option
                            className=""
                            disabled={false}
                            value="bar"
                          >
                            Bar
                          </option>
                        </FormSelectOption>
                      </select>
                    </FormSelect>
                  </Select>
                </div>
              </FormGroup>
            </Pf4SelectWrapper>
          </div>
        </GridItem>
      </div>
    </Grid>
  </Route>
</ItemDetailDescription>
`;
